### PR TITLE
SharedMemoryMapTests: optimize GetStream_VerifyContentMatches

### PR DESCRIPTION
**Pull request by NickCraver**
The comparison here was an implementation we can optimize lowering the test runtime by reducing allocations, copies, and a suboptimal comparison (.NET 6 further optimizes sequence comparisons). I don't see a quick way to optimize it further that doesn't compromise what we want to test, so stopping here. This is out of an 8 min runtime locally for all test (I don't have everything configured) so overall decent savings for test runs.

If you're looking just above these changes at:
```cs
[InlineData(1024 * 1024 * 1024, Skip = "Results in OOM when run on test infra")] // 1GB
```
...then why doesn't the 1GB test skip locally? I'm not sure, but it runs which is why I looked at this. I'll poke at that, but it still helps shave off the 256MB even with it fixed later.

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

Before:
image
After:
image

### CODE REVIEW

**Comment by NickCraver** [comment]
Test failures aren't related - random flakiness both runs.

**Review by gohar94 — DISMISSED** [review]
Thanks, Nick.

File: test/WebJobs.Script.Tests/Workers/SharedMemoryMapTests.cs [hunk 1]
```diff
@@ -209,14 +211,17 @@ public async Task GetStream_VerifyContentMatches(int contentSize)
 
             Stream readContent = await sharedMemoryMap.GetStreamAsync();
 
-            // Read content into MemoryStream to easily put them into a byte[].
-            using (MemoryStream tempStream = new MemoryStream())
+            static IEnumerable<byte> StreamToEmulerable(Stream stream)
```
Comment (surgupta-msft) [inline test/WebJobs.Script.Tests/Workers/SharedMemoryMapTests.cs]: nit: Enumerable typo in StreamToEmulerable

**Review by surgupta-msft — DISMISSED** [review]

File: test/WebJobs.Script.Tests/Workers/SharedMemoryMapTests.cs (hunk 1 above)
Comment (NickCraver) [reply test/WebJobs.Script.Tests/Workers/SharedMemoryMapTests.cs]: I...don't know what happened there :( Fixed!

**Review by gohar94 — APPROVED** [review]

**Review by surgupta-msft — APPROVED** [review]

**Pull request merged**